### PR TITLE
Fix crm_resource --validate validation

### PR DESCRIPTION
**Pull request by nrwahl2**
In case of an error, the output of `crm_resource --validate` fails to validate (if validate-all does not output XML). This is because if a
`<choice>` contains two elements with the same name in a RelaxNG schema, only the first occurrence is honored and the rest are ignored. (This does not seem to be documented clearly; it's a conclusion based on experimentation.)

A solution is to create just one `<element>` that contains a `<choice>` (instead of a `<choice>` that contains two `<element>`s).

Closes RHBZ#2123727

### CODE REVIEW

**Comment by nrwahl2** [comment]
```
* faiiled (rc=064): crm_resource   - Validate a valid resource configuration
* faiiled (rc=064): crm_resource   - Validate a valid resource configuration (XML)
* faiiled (rc=064): crm_resource   - Validate an invalid resource configuration
* faiiled (rc=064): crm_resource   - Validate an invalid resource configuration (XML)
```

Bizarre, with the misspelling, and doesn't happen on my machine with the RPM target. Let's try again...

File: xml/Makefile.am [hunk 1]
```diff
@@ -69,7 +69,16 @@ API_request_base	= command-output	\
 CIB_cfg_base		= options nodes resources constraints fencing acls tags alerts
 
 # Names of all schemas (including top level and those included by others)
-API_base		= $(API_request_base) fence-event failure generic-list item node-attrs node-history nodes resources status
+API_base		= $(API_request_base)	\
+			  fence-event failure	\
```
Comment (kgaillot) [inline xml/Makefile.am:73]: did you mean to put these on separate lines?

**Review by kgaillot — COMMENTED** [review]
looks good

**Comment by kgaillot** [comment]
> ```
> * faiiled (rc=064): crm_resource   - Validate a valid resource configuration
> * faiiled (rc=064): crm_resource   - Validate a valid resource configuration (XML)
> * faiiled (rc=064): crm_resource   - Validate an invalid resource configuration
> * faiiled (rc=064): crm_resource   - Validate an invalid resource configuration (XML)
> ```
> 
> This misspelling is odd.

What's odd is I can't find where it happens, but this is a hack to make expected failures not hit a grep for fail.

> Anyway, the new regression tests failed in CI because the `ocf:pacemaker:Dummy` and `ocf:heartbeat:IPaddr2` resource agents don't exist there. Any advice on how to deal with that?

Avoid IPaddr2; modify Dummy if needed. For Dummy, not just CI but source checkout runs will be affected as well. That's a conundrum since the RA search path is baked in at configure time. While we could add a source directory to the RA path for CI purposes, I don't think we'd want to require that in a source checkout in order run cts-cli.

I'm thinking we should move these tests to a new "agents" test group in cts-cli, and not run it by default. The new tests would need to be run in an installed environment, and wouldn't be run by CI, but we could run them manually at release time to check for regressions.

**Comment by kgaillot** [comment]
> What's odd is I can't find where it happens, but this is a hack to make expected failures not hit a grep for fail.

Ah now I remember, it's in pacemaker.spec.in

File: xml/Makefile.am (hunk 1 above)
Comment (nrwahl2) [reply xml/Makefile.am:73]: Yes, thought I caught all of those, thanks

**Comment by nrwahl2** [comment]
Updated to (hopefully) address the CI test failures.

**Review by kgaillot — APPROVED** [review]
Awesome, thanks

**Comment by nrwahl2** [comment]
forgot to remove the validate stuff from the tools.exp

**Pull request merged**